cmd/contract_cmd.go

```diff
@@ -0,0 +1,188 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/ontio/ontology/account"
+	"github.com/ontio/ontology/cmd/utils"
+	"github.com/ontio/ontology/common"
+	"github.com/ontio/ontology/smartcontract/types"
+	"github.com/urfave/cli"
+)
+
+var (
+	ContractCommand = cli.Command{
+		Name:         "contract",
+		Action:       utils.MigrateFlags(contractCommand),
+		Usage:        "ontology contract [invoke|deploy] [OPTION]",
+		Category:     "CONTRACT COMMANDS",
+		OnUsageError: contractUsageError,
+		Description:  `account command`,
+		Subcommands: []cli.Command{
+			{
+				Action:       utils.MigrateFlags(invokeContract),
+				Name:         "invoke",
+				OnUsageError: invokeUsageError,
+				Usage:        "ontology invoke [OPTION]\n",
+				Flags:        append(NodeFlags, ContractFlags...),
+				Category:     "CONTRACT COMMANDS",
+				Description:  ``,
+			},
+			{
+				Action:       utils.MigrateFlags(deployContract),
+				OnUsageError: deployUsageError,
+				Name:         "deploy",
+				Usage:        "ontology deploy [OPTION]\n",
+				Flags:        append(NodeFlags, ContractFlags...),
+				Category:     "CONTRACT COMMANDS",
+				Description:  ``,
+			},
+		},
+	}
+)
+
+func contractCommand(ctx *cli.Context) error {
+	showContractHelp()
+	return nil
+}
+
+func contractUsageError(context *cli.Context, err error, isSubcommand bool) error {
+	fmt.Println(err.Error())
+	showContractHelp()
+	return nil
+}
+
+func invokeUsageError(context *cli.Context, err error, isSubcommand bool) error {
+	fmt.Println(err.Error())
+	showInvokeHelp()
+	return nil
+}
+
+func invokeContract(ctx *cli.Context) error {
+	if !ctx.IsSet(utils.ContractAddrFlag.Name) || !ctx.IsSet(utils.ContractParamsFlag.Name) {
+		showInvokeHelp()
+		return nil
+	}
+
+	client := account.GetClient(ctx)
+	if client == nil {
+		log.Fatal("Can't get local account.")
+		os.Exit(1)
+	}
+
+	acct := client.GetDefaultAccount()
+
+	contractAddr := ctx.String(utils.ContractAddrFlag.Name)
+	params := ctx.String(utils.ContractParamsFlag.Name)
+	if "" == contractAddr {
+		log.Fatal("contract address does not allow empty")
+	}
+
+	addr, _ := common.AddressFromBase58(contractAddr)
+	txHash, err := ontSdk.Rpc.InvokeNeoVMSmartContract(acct, new(big.Int), addr, []interface{}{params})
+	if err != nil {
+		log.Fatalf("InvokeSmartContract InvokeNeoVMSmartContract error:%s", err)
+	} else {
+		fmt.Printf("invoke transaction hash:%s", common.ToHexString(txHash[:]))
+	}
+
+	//WaitForGenerateBlock
+	_, err = ontSdk.Rpc.WaitForGenerateBlock(30*time.Second, 1)
+	if err != nil {
+		log.Fatalf("InvokeSmartContract WaitForGenerateBlock error:%s", err)
+	}
+	return nil
+}
+
+func getVmType(vmType uint) types.VmType {
+	switch vmType {
+	case 0:
+		return types.Native
```

Comment:
there has no way to deploy native contract, could please remove it ,thanks.

Reply:
Thx. have removed.